### PR TITLE
fix(agw): Fixed compilation error on inclusion of proto_map.h file 

### PR DESCRIPTION
**Pull request by rsarwad**
## Summary
fix(agw): Fixed compilation error on inclusion of proto_map.h file  and removed unused file, sgw.h and unused structure, predefined_pcc_rules 

## Test Plan
Executed s1ap sanity test suite and unit test cases

### CODE REVIEW

**Review by ajahl — APPROVED** [review]

File: lte/gateway/c/core/oai/test/spgw_task/spgw_procedures_test_fixture.cpp [hunk 1]
```diff
@@ -24,6 +24,7 @@ extern "C" {
 #include "lte/gateway/c/core/oai/include/gx_messages_types.h"
 #include "lte/gateway/c/core/oai/include/ip_forward_messages_types.h"
 #include "lte/gateway/c/core/oai/lib/3gpp/3gpp_23.401.h"
+#include "lte/gateway/c/core/oai/tasks/sgw/sgw_defs.hpp"
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/test/spgw_task/spgw_procedures_test_fixture.cpp:27]: This should be outside `extern "C"` block

File: lte/gateway/c/core/oai/tasks/sgw/spgw_state_converter.cpp [hunk 2]
```diff
@@ -16,11 +16,11 @@
  */
 
 #include "lte/gateway/c/core/oai/tasks/sgw/spgw_state_converter.hpp"
+#include "lte/gateway/c/core/oai/include/sgw_context_manager.hpp"
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/tasks/sgw/spgw_state_converter.cpp]: Move this after the `extern "C"` block as per [style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

File: lte/gateway/c/core/oai/include/sgw_context_manager.hpp [hunk 3]
```diff
@@ -23,15 +23,18 @@
  */
 #pragma once
 
+#include <stdint.h>
+
+#include "lte/gateway/c/core/oai/include/spgw_state.hpp"
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/include/sgw_context_manager.hpp]: This should be after the `extern "C"` block as per the [style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

File: lte/gateway/c/core/oai/include/sgw_s8_state.hpp [hunk 4]
```diff
@@ -13,14 +13,18 @@ limitations under the License.
 
 #pragma once
 
+#include <pthread.h>
+
+#include "lte/gateway/c/core/oai/include/spgw_types.hpp"
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/include/sgw_s8_state.hpp]: This should be after the `extern "C"` block as per the [style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

File: lte/gateway/c/core/oai/include/spgw_state.hpp [hunk 5]
```diff
@@ -17,16 +17,16 @@
 
 #pragma once
 
+#include <pthread.h>
+
+#include "lte/gateway/c/core/oai/include/spgw_types.hpp"
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/include/spgw_state.hpp]: This should be after the `extern "C"` block as per the [style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

File: lte/gateway/c/core/oai/include/spgw_types.hpp [hunk 6]
```diff
@@ -30,10 +30,17 @@
 #ifndef FILE_SPGW_TYPES_SEEN
 #define FILE_SPGW_TYPES_SEEN
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/include/spgw_types.hpp]: Please replace with `#pragma once`

File: lte/gateway/c/core/oai/lib/mobility_client/MobilityClientAPI.hpp [hunk 7]
```diff
@@ -20,13 +20,14 @@
 
 #include "stdint.h"
 
+#include "lte/gateway/c/core/oai/include/spgw_state.hpp"
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/lib/mobility_client/MobilityClientAPI.hpp]: This should be after the C headers as per the [style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

File: lte/gateway/c/core/oai/lib/pcef/pcef_handlers.cpp [hunk 8]
```diff
@@ -15,29 +15,30 @@
...
+
 #include <string>
 
+#include <grpcpp/impl/codegen/status.h>
+#include "lte/protos/session_manager.pb.h"
+
+#include "lte/gateway/c/core/oai/lib/pcef/PCEFClient.hpp"
+#include "lte/gateway/c/core/oai/lib/mobility_client/MobilityClientAPI.hpp"
+#include "lte/gateway/c/core/oai/include/spgw_types.hpp"
+
```
Comment (ssanadhya) [inline lte/gateway/c/core/oai/lib/pcef/pcef_handlers.cpp]: This should be after the `extern "C"` block as per the [style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

**Review by ssanadhya — CHANGES_REQUESTED** [review]
Please review the order of includes in [style guide]((https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)

**Review by ssanadhya — APPROVED** [review]
Lgtm

**Comment by ssanadhya** [comment]
@rsarwad , could you please resolve CI failures on this PR?

**Comment by rsarwad** [comment]
Hi @ssanadhya, the signoffing of an commit was missing on intermediate commit. Fixing it lead to some other errors. So closing this PR and raised new PR, https://github.com/magma/magma/pull/13833

**Pull request closed without merging**